msal/src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -282,7 +282,8 @@ public async Task RemoveAsync(IAccount account)
                 authority = Core.Instance.Authority.CreateAuthority(Authority, ValidateAuthority);
                 var tenantId = authority.GetTenantId();
 
-                if (Core.Instance.Authority.TenantlessTenantNames.Contains(tenantId))
+                if (Core.Instance.Authority.TenantlessTenantNames.Contains(tenantId) 
+                    && account.HomeAccountId?.TenantId != null)
```

Comment:
I tried this out w/a null account (silent flow, no cache) and got a null reference here because the account object is null. We check for a null account next in the SilentRequest and throw an MsalUiRequiredException there. we need to ensure account is not null before moving forward.

Reply:
account is mandatory parameter for this api